packages/tools/client-debug-view/src/components/ContainerSelectionDropdown.tsx

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Dropdown, IDropdownOption, IDropdownStyles, IStackTokens, Stack } from "@fluentui/react";
+import React from "react";
+
+import {
+	DebuggerRegistry,
+	IFluidClientDebugger,
+	getDebuggerRegistry,
+	getFluidClientDebuggers,
+} from "@fluid-tools/client-debugger";
+
+import { HasClientDebugger } from "../CommonProps";
+
+/**
+ * {@link ContainerSelectionDropdown} input props.
+ */
+export type ContainerSelectionDropdownProps = HasClientDebugger;
+
+/**
+ * Small header that displays core container data.
+ *
+ * @param props - See {@link ContainerSelectionDropdownProps}.
+ */
+export function ContainerSelectionDropdown(
+	props: ContainerSelectionDropdownProps,
+): React.ReactElement {
+	const dropdownStyles: Partial<IDropdownStyles> = {
+		dropdown: { width: "300px", zIndex: "1" },
+	};
+
+	const stackTokens: IStackTokens = { childrenGap: 20 };
+
+	const debuggerRegistry: DebuggerRegistry = getDebuggerRegistry();
+
+	function renewContainerOptions(debuggers: IFluidClientDebugger[]): IDropdownOption[] {
+		const options: IDropdownOption[] = [];
+		for (const x of debuggers) {
+			console.log(x.containerId);
+			options.push({
+				key: x.containerId,
+				text: x.containerNickname ?? x.containerId,
+			});
+		}
+		return options;
+	}
+
+	const [clientDebuggers, setClientDebuggers] = React.useState<IFluidClientDebugger[]>(
+		getFluidClientDebuggers(),
+	);
+
+	const [clientDebuggerOptions, setClientDebuggerOptions] = React.useState<IDropdownOption[]>(
+		renewContainerOptions(clientDebuggers),
+	);
+
+	React.useEffect(() => {
+		function onDebuggerChanged(): void {
+			setClientDebuggers(getFluidClientDebuggers());
+			setClientDebuggerOptions(renewContainerOptions(clientDebuggers));
+		}
+
+		debuggerRegistry.on("debuggerRegistered", onDebuggerChanged);
+		debuggerRegistry.on("debuggerClosed", onDebuggerChanged);
+
+		return (): void => {
+			debuggerRegistry.off("debuggerRegistered");
+			debuggerRegistry.off("debuggerClosed");
+		};
+	}, [clientDebuggers, clientDebuggerOptions, setClientDebuggers, setClientDebuggerOptions]);
+
+	const _onClientDebuggerDropdownChange = (
+		event: React.FormEvent<HTMLDivElement>,
+		option?: IDropdownOption,
+	): void => {
+		if (option) {
+			const selectedDebugger = clientDebuggers.find((c) => {
+				return c.containerId === (option.key as string);
+			}) as IFluidClientDebugger;
+			props.clientDebugger = selectedDebugger;
```

Comment:
This won't update the container selection from the perspective of `ClientDebugView`. In order to get the parent container to change its selection based on the drop-down, we probably need to have `ClientDebugView` manage the list of debuggers and the current selection and update this component's props to take in the list of debuggers, the current selection, and an `onSelectionChange` (or something like that) callback to inform `ClientDebugView` when the selection changes.

Reply:
Adding "currentDisplayDebugger" in register and make it updating in debugger.tsx